core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/NodePhysicalComponentExchangesCreator.java

```diff
@@ -135,9 +133,14 @@ protected void createExchangesForPhysicalComponents(PhysicalComponent containerP
       }
     }
     // Process contained Physical Actors
-    containerPC.getOwnedPhysicalComponents().stream().filter(c -> ComponentExt.isActor(c)).forEach(actor -> {
-      createExchangesForPhysicalComponents(containerPC, actor);
-    });
+    if (!containerPC.getOwnedPhysicalComponents().isEmpty()) {
+      containerPC.getOwnedPhysicalComponents().stream().filter(c -> ComponentExt.isActor(c)).forEach(actor -> {
+        createExchangesForPhysicalComponents(containerPC, actor);
+      });
+    } else if (ComponentExt.isActor(containerPC)){
+      createExchangesForPhysicalComponents(containerPC, containerPC);

```

Comment:
Is this needed? Since PhysicalLinkCreator will handle the case where the container is an Actor.